bc-rest/src/main/java/org/jboss/da/bc/facade/AbstractBuildConfigurationsFacade.java

```diff
@@ -140,8 +144,15 @@ protected void fillInfoEntity(InfoEntity ie, GeneratorEntity ge) {
 
     protected <T extends GeneratorEntity> T toGeneratorEntity(EntityConstructor<T> constructor,
             InfoEntity ie) {
-        String url = ie.getTopLevelBc().getScmUrl();
-        String revision = ie.getTopLevelBc().getScmRevision();
+        String url = "";
+        String revision = "";
+        if (ie.getTopLevelBc().getScmUrl() == null) {
+            url = ie.getTopLevelBc().getExternalScmUrl();
+            revision = ie.getTopLevelBc().getExternalScmRevision();
+        } else {
+            url = ie.getTopLevelBc().getScmUrl();
+            revision = ie.getTopLevelBc().getScmRevision();
+        }
```

Comment:
This will always set ``GeneratorEntity.toplevelBc.scmUrl`` and ``GeneratorEntity.toplevelBc.scmUrl`` (ie the internal scm coordinates), regadless of the fact that ``InfoEntity.topLevelBc`` could have internal or external coordinates.